src/Centreon/Domain/Engine/EngineConfiguration.php

```diff
@@ -141,11 +143,10 @@ public function setName(?string $name): EngineConfiguration
      */
     public function removeIllegalCharacters(string $stringToAnalyse): string
     {
-        if ($this->illegalObjectNameCharacters === null) {
+        if ($this->illegalObjectNameCharacters === null || $this->illegalObjectNameCharacters === '') {
```

Comment:
```suggestion
        if (empty($this->illegalObjectNameCharacters)) {
```

Reply:
It works, I am ok with this suggestion, but there are schools which prefer explicit type check.
I could have written also like that (more restrictive than empty) : 

```
if ('' === (string) $this->illegalObjectNameCharacters)
```